pkg/manager/runnable_group_test.go

```diff
@@ -161,6 +161,42 @@ var _ = Describe("runnableGroup", func() {
 		}
 	})
 
+	It("should be able to handle adding runnables while stopping", func() {
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+		rg := newRunnableGroup(defaultBaseContext, errCh)
+
+		go func() {
+			defer GinkgoRecover()
+			<-time.After(1 * time.Millisecond)
+			Expect(rg.Start(ctx)).To(Succeed())
+		}()
+		go func() {
+			defer GinkgoRecover()
+			<-time.After(1 * time.Millisecond)
+			ctx, cancel := context.WithCancel(context.Background())
+			cancel()
+			rg.StopAndWait(ctx)
```

Comment:
There is no guarantee that this is called after rg.Start, right? Is this intended?

If I see correctly it doesn't really matter which one is called first as StopAndWait calls Start internally and Start is using once? (just to confirm)

Reply:
That's correct, it doesn't really matter because they can both run at the same time, the waitgroup triggers. It's a best effort to try to reproduce the failure consistently, but not a guaranteed one. Although on my mac, it did trigger the race pretty consistently